### ISSUE TITLE
ERNEST v1.11.0 - AuthFailure

### ISSUE THREAD

**Issue by asantos82** [ernestio/ernest#225]
After upgrading to Ernest v1.11.0 for Ernest and Ernest-CLI I am unable to login to AWS.
In v1.10.0 it works OK.

Bellow you can find the error, and in attachment the logs

```
ernest-cli-v11 target https://127.0.0.1
Target set
ernest-cli-v11 login --user admin --password admin
Welcome back admin
ernest-cli-v11 user create user1 abc123
User user1 successfully created
ernest-cli-v11 group create group1
Group 'group1' successfully created, you can add users with 'ernest group add-user username group1'
ernest-cli-v11 group add-user user1 group1
User 'user1' is now assigned to group 'group1'
ernest-cli-v11 login --user user1 --password abc123
Welcome back user1
ernest-cli-v11 datacenter create aws --region eu-west-1 --secret_access_key iYxxxxxxxxxxxxxxxxxxxxxxxxxxxqe --access_key_id AYYYYYYYYYYYYYYYYYYYQ my-dc
Datacenter 'my-dc' successfully created 

ernest-cli-v11 service apply demo.yml
Environment creation requested
Ernest will show you all output from your requested service creation
You can cancel at any moment with Ctrl+C, even the service is still being created, you won't have any output
Applying you definition
Creating networks:
 - my-dc-demo-public
   IP     : 10.0.10.0/24
   Status : errored
   Error     : AuthFailure: AWS was not able to validate the provided access credentials
	status code: 401, request id: 14f81ce8-6e3f-43cd-b53e-d830cae28b53
 - my-dc-demo-private
   IP     : 10.0.11.0/24
   Status : errored
   Error     : AuthFailure: AWS was not able to validate the provided access credentials
	status code: 401, request id: a332cf26-5d32-44fb-9ca4-d912f245526c
Networks creation failed

Oops! Something went wrong. Please manually fix any errors shown above and re-apply your definition.
```

[ernest.log.zip](https://github.com/ernestio/ernest/files/680954/ernest.log.zip)

**Comment by purehyperbole** [comment]
Thanks for the logs! This is likely the result of the new encryption we are using for datacenter credentials (introduced in 1.11.0). I've pushed a fix to master as it looks like some environment variables were missing. Please could you run setup again with the new docker-compose.yml and re-test?

Please note, you may need to re-add your aws credentials using:

ernest datacenter update aws --secret_access_key <secret_access_key> --access_key_id <access_key_id> my_datacenter

**Comment by asantos82** [comment]
HI,

The v1.11.0 is a new setup. I made the setup from ground up, and got this issue. The I have rolled-back to v1.10.0 and set it all up again and it worked.
Every time I update docker images to v1.11.0 and ernest-cli to v1.11.0 and setup a datacenter I get that error.

I tried to perform what you've suggested and I get another error. As far as I can see something is wrog with the parsing of char "/"

```
ernest-cli-v11 service apply demo.yml
Environment creation requested
Ernest will show you all output from your requested service creation
You can cancel at any moment with Ctrl+C, even the service is still being created, you won't have any output
Applying you definition
Creating networks:
 - my-dc-demo-public
   IP     : 10.0.10.0/24
   Status : errored
   Error     : AuthFailure: Credential must have exactly 5 slash-delimited elements, e.g. keyid/date/region/service/term, got '5NXPY8qVifT6081t3U7j46AAyShSUFsmwJm97FLuWp9cnuBbfrTB/BaaYiU=/20170103/eu-west-1/ec2/aws4_request,'
	status code: 401, request id: c61b79a1-5197-40c7-9c06-4be76cdf4a0a
 - my-dc-demo-private
   IP     : 10.0.11.0/24
   Status : errored
   Error     : AuthFailure: Credential must have exactly 5 slash-delimited elements, e.g. keyid/date/region/service/term, got '5NXPY8qVifT6081t3U7j46AAyShSUFsmwJm97FLuWp9cnuBbfrTB/BaaYiU=/20170103/eu-west-1/ec2/aws4_request,'
	status code: 401, request id: d1809862-35c9-4b72-bd67-95b6f7fb409e
Networks creation failed

Oops! Something went wrong. Please manually fix any errors shown above and re-apply your definition.
```

I have checked the table datacenters in DB, and both the access_key_id and the secret_access_key have the char "/"

**Comment by purehyperbole** [comment]
Hi, 

After some testing, I'm unable to re-create this exact issue with a clean deploy of 1.11.0 or an upgrade from 1.10.0 to 1.11.0. During testing i did notice incompatibilities between 1.10.0 and 1.11.0 CLI versions, so if you have upgraded from 1.10.0, its worth checking that the cli has been bumped to 1.11.0 before running 'ernest-cli datacenter update'

Are the credentials stored on the database that you mention stored in plain text?  Also, could you confirm that a sensible value has been set for 'ERNEST_CRYPTO_KEY' in the docker-compose.yml file?

If you are seeing slashes in credential values, it's most likely an encrypted string. This then gets decrypted before it's used.

**Comment by asantos82** [comment]
Hi,

The docker-compose.yml has been updated just 22hours ago.
I'll try with this commit, and then I'll give you some feedback

**Comment by asantos82** [comment]
Hi,

This issue has been solved in commit e90ee738a87d7b7e5156a7f65426d157780a0131

**Comment by adriacidre** [comment]
Cheers @asantos82 , I'll close this issue if it is fixed :)
